eo-parser/src/main/java/org/eolang/parser/Syntax.java

```diff
@@ -110,41 +115,66 @@ public Syntax(
     /**
      * Compile it to XML and save.
      *
+     * <p>No exception will be thrown if the syntax is invalid. In any case, XMIR will
+     * be generated and saved. Read it in order to find the errors,
+     * at <tt>/program/errors</tt> XPath.</p>
+     *
      * @throws IOException If fails
+     * @checkstyle AnonInnerLengthCheck (100 lines)
      */
     public void parse() throws IOException {
         final List<Text> lines = this.lines();
-        final ANTLRErrorListener errors = new BaseErrorListener() {
+        final List<ParsingException> errors = new LinkedList<>();
+        final ANTLRErrorListener spy = new BaseErrorListener() {
```

Comment:
@yegor256 [Suggestion]: Maybe it's better to create a separate `private` class? Something like `SpyErrorListener`:
```java
private class SpyErrorListener extends BaseErrorListener{

@Override
public void syntaxError(...);

public boolean hasErrors();

public XML directives();

}
```
It is only the sketch, of course. By this you:

- Don't need to specify `@checkstyle AnonInnerLengthCheck (100 lines)`
- Will simplify the `parse` method significantly

Reply:
@volodya-lombrozo you are right, fixed: 58f82c49f